### DISCUSSION TITLE
Unable to insert with row level security enabled

### DISCUSSION THREAD

**Discussion by christian-valadez** [supabase/supabase#1548] [Questions] [answered]
I'm having some trouble writing to my database with policies enabled. 

![image](https://user-images.githubusercontent.com/22481651/118017857-eef55600-b30b-11eb-9c49-55a90cb1f203.png)

I have an accounts table, and one of the columns for that table is user_id. For selects, updates, and deletes, I'd like only users to be able to view and mutate their own data, so I have a policy where `uid() = user_id`. 

For inserts, any authenticated user should be able to add data, so I set the policy to just be `true`. But, I'm unable to insert - I get this error: 

```
 code: '42501',
  message: 'new row violates row-level security policy for table "accounts"'
```

Any insight / help into this would be very much appreciated!

**Comment by easy-street** [comment, marked as answer]
as described in the [insert docs](https://supabase.io/docs/reference/javascript/insert#notes), this is because the client library makes a `select` when inserting to return the full record, applying the RLS policies for `select`.  the `returning` param can be provided with `options` to prevent this:

```
await supabase
  .from("accounts")
  .insert({ email: "email@example.com" }, { returning: "minimal" });
```

note that the `true` policy for insert will allow anonymous users to insert as well. to allow only authenticated users, use:
```
auth.role() = 'authenticated'
```

**Reply by christian-valadez** [reply]
whoops, looks like i missed that part of the docs. Thanks for your help, and for the pointer on the authenticated role! 🙏
